WcaOnRails/app/assets/javascripts/competitions.js

```diff
@@ -99,3 +99,32 @@ onPage('competitions#index', function() {
     }, 0);
   });
 });
+
+onPage("competitions#show_all_results", function() {
+  /* Show tbody for the given event. */
+  function showResultsFor(eventId) {
+    var $results = $('.one-event');
+    $results.hide();
+    $results.filter('.event-' + eventId).show();
+  }
+
+  /* Handle events selector for results by event. */
+  $('.event-selector input[type="radio"]').on('change', function() {
+    var eventId = $(this).val();
+    showResultsFor(eventId);
+    $.setUrlParams({ event:  eventId });
+  });
+
+  if(location.hash) {
+    /* Support old URLs with hash indicating an event id. */
+    $('.event-selector #radio-' + location.hash.slice(2)).click();
```

Comment:
Does this open the possibility for an injection attack? I think it's possible to create elements, which can cause arbitrary javascript to run... See http://www.mjcblog.net/2011/06/jquery-selector-injection/ for more information. It sounds like jquery may have changed to protect against this particular attack, but even if it has, I think it would feel more comfortable using something borking like `document.getElementById`, or sanitizing the input to make sure it's a valid event id.

nit: It was not clear to me what the `slice(2)` is doing here. I guess this is because the anchors look like `#e444`, and this to strip out the `#e`? I think it would be nice to document this clearly, especially since the code that generates these links will be disappearing.

Reply:
Ahh, yikes, hadn't ever thought about injection via jquery.  Switched to plain JS.

Added a comment about the hash format.